src/test/java/net/snowflake/client/jdbc/SnowflakeDriverLatestIT.java

```diff
@@ -61,6 +65,36 @@ private boolean isFileContentEqual(
     }
   }
 
+  @Test
+  public void testClientInfoConnectionProperty() throws Throwable {
+    Connection connection = null;
+    Statement statement = null;
+    ResultSet res = null;
+
+    try {
+      Properties props = new Properties();
+      props.put("snowflakeClientInfo", "{\"sparkVersion\":\"1.2.0\", \"sparkApp\":\"mySparkApp\"}");
+      connection = getConnection(DONT_INJECT_SOCKET_TIMEOUT, props, false, false);
+
+      statement = connection.createStatement();
+
+      res = statement.executeQuery("select current_session_client_info()");
+
+      assertTrue("result expected", res.next());
+
+      String clientInfoJSONStr = res.getString(1);
+
+      JsonNode clientInfoJSON = mapper.readTree(clientInfoJSONStr);
+
+      // assert that spart version and spark app are found
+      assertEquals("spark version mismatch", "1.2.0", clientInfoJSON.get("sparkVersion").asText());
+
+      assertEquals("spark app mismatch", "mySparkApp", clientInfoJSON.get("sparkApp").asText());
+    } finally {
+      closeSQLObjects(res, statement, connection);
+    }
+  }
+
```

Comment:
Could you please add another 2 test case:
1. Test System.setPropertye() also works like JDBC property.
2. Both System.setPropertye() and JDBC property are set, assert the option set by  JDBC property is used actually.

Reply:
Done

Reply:
Added Test #2 to this same test below the current test.

Test#1, testing system.setproperty(), is already tested in SnowflakeDriverIT. I changed it slightly.